### PR TITLE
[#7685] Fix flex warning for GenQuery2 lexer rules (4-3-stable)

### PR DESCRIPTION
**Pull request by korydraughn**
All unit/core tests pass.

### CODE REVIEW

**Review by alanking — COMMENTED** [review]
What is the result of the commit with no issue number? Will that need a test?

**Comment by korydraughn** [comment]
I haven't decided on whether to create a new issue or link it to the housekeeping one yet.

**Comment by korydraughn** [comment]
Added issue number to commit.

**Review by alanking — COMMENTED** [review]
Just to make sure... Should we add a test for the multi-whitespace capturing change? Other than this question, changes look good to me.

**Comment by korydraughn** [comment]
I don't think we need one, but it doesn't hurt to add a test for that.

Adding test now.

**Comment by korydraughn** [comment]
Test added and it passes.

File: unit_tests/src/test_rc_genquery2.cpp [hunk 1]
```diff
@@ -118,4 +118,30 @@ TEST_CASE("rc_genquery2")
...
+
+        		where       	COLL_NAME
+=      				'{}')", env.rodsHome);
+
+        Genquery2Input input{};
+        input.query_string = query_string.data();
+
+        char* output{};
+        REQUIRE(rc_genquery2(static_cast<RcComm*>(conn), &input, &output) == 0);
+        REQUIRE_FALSE(output == nullptr);
```
Comment (alanking) [inline unit_tests/src/test_rc_genquery2.cpp:139]: Will `output` leak once we get past this assertion?

Comment (korydraughn) [reply unit_tests/src/test_rc_genquery2.cpp:139]: It sure will.

Nice catch.

Will fix this one and check the other tests too.

Comment (korydraughn) [reply unit_tests/src/test_rc_genquery2.cpp:160]: Fixed.

**Comment by alanking** [comment]
Looks good to me... Once we've got an issue number for that last commit, please squash to taste.

**Comment by korydraughn** [comment]
Tests still pass.

Squashing.

**Comment by korydraughn** [comment]
Squashed.

**Review by alanking — APPROVED** [review]
\# it

**Comment by korydraughn** [comment]
Pounded.
